app/src/org/commcare/activities/LoginActivity.java

```diff
@@ -238,11 +238,22 @@ protected boolean checkForSeatedAppChange() {
         String currentSeatedId = CommCareApplication.instance().getCurrentApp().getUniqueId();
         if (!lastSeatedId.equals(currentSeatedId)) {
             prefs.edit().putString(KEY_LAST_APP, currentSeatedId).commit();
+            disableWorkForLastSeatedApp();
             return true;
         }
         return false;
     }
 
+
+    // cancels all worker tasks for previously seated app
+    private static void disableWorkForLastSeatedApp() {
+        SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(CommCareApplication.instance());
+        String lastSeatedId = prefs.getString(KEY_LAST_APP, "");
+        if(!lastSeatedId.isEmpty()) {
+            WorkManager.getInstance(CommCareApplication.instance()).cancelAllWorkByTag(lastSeatedId);

```

Comment:
Quick note that this request isn't synchronous. I'm not sure how killing the update tasks worked when we unseated apps, but if you haven't checked it would be worth confirming that the semantics are the same.

Reply:
From the [note here](https://github.com/dimagi/commcare-android/blob/master/app/src/org/commcare/engine/resource/AndroidResourceManager.java#L280-L281), it looks like instead of cancelling the auto update for previous app, we were just firing the update request for the new seated app during the auto update process. We cancel all running tasks today [on logout](https://github.com/dimagi/commcare-android/blob/master/app/src/org/commcare/CommCareApplication.java#L322) which takes care of update request getting cancelled on logout and then when the user logins, we reinititate the task for the current app. We can follow the same semantics for cancelling the WorkManager tasks though since app updates can work fine even when the user is logged out, we can choose to just cancel work on app seating changes to keep the last update downloads running even when user is logged out.